CRM/Utils/Date.php

```diff
@@ -161,20 +161,25 @@ public static function format($date, $separator = '', $invalidDate = 0) {
    */
   public static function getAbbrWeekdayNames() {
     $key = 'abbrDays_' . \CRM_Core_I18n::getLocale();
-    $days = &\Civi::$statics[__CLASS__][$key];
-    if (!$days) {
-      $days = [];
+    if (empty(\Civi::$statics[__CLASS__][$key])) {
+      $days = [
+        0 => ts('Sun'),
+        1 => ts('Mon'),
+        2 => ts('Tue'),
+        3 => ts('Wed'),
+        4 => ts('Thu'),
+        5 => ts('Fri'),
+        6 => ts('Sat'),
+      ];
```

Comment:
Throwing out an idea (not sure it's a better idea, just _an_ idea, and would solve the chicken/egg with the test and reduce the burden on translators): `intl` includes translations for all these things without needing to fiddle with OS locale packages. `ts` is good because it "understands civi", and bypassing it would bypass any custom translation function you've set, but these are standard things so the benefit of using intl may outweigh that downside.

So
```php
 $intl_formatter = IntlDateFormatter::create(CRM_Core_I18n::getLocale(), IntlDateFormatter::MEDIUM, IntlDateFormatter::MEDIUM, null, IntlDateFormatter::GREGORIAN, 'E');
 $days = [
        0 => $intl_formatter->format(strtotime('Sunday')),
        1 => $intl_formatter->format(strtotime('Monday')),
        2 => $intl_formatter->format(strtotime('Tuesday')),
        3 => $intl_formatter->format(strtotime('Wednesday')),
        4 => $intl_formatter->format(strtotime('Thursday')),
        5 => $intl_formatter->format(strtotime('Friday')),
        6 => $intl_formatter->format(strtotime('Saturday')),
      ];
```

For the long day format it's `EEEE` instead, but otherwise identical. For month names it would be `strtotime('January')` etc, and it's `MMM` for short and `MMMM` for long.

`ts()` may have some languages that `intl` doesn't, but I tried a range of languages and it seems pretty good.

Reply:
@mlutfy thoughts?

Reply:
Yeah - the only question is - does everyone always have `intl` - we could maybe require it for php8.1 (even if only in words) and fall back to what we do currently if not present?

Reply:
I think relying on `php-intl` is a good idea. We already rely on less-frequently used packages, such as `bcmath`, and `intl` is pretty much always installed on systems I've seen.

And `php-intl` indeed does not seem to require the locale to be enabled by the OS. I tested locally on a system without the `uk_UA` locale:

```
$intl_formatter = IntlDateFormatter::create('uk_UA', IntlDateFormatter::MEDIUM, IntlDateFormatter::MEDIUM, null, IntlDateFormatter::GREGORIAN, 'E');
echo $intl_formatter->format(strtotime('Monday')) . "\n";
```

outputs: `пн`, which is correct.